packages/api/src/webhooks/cannon.ts

```diff
@@ -52,7 +52,70 @@ export default class WebhookCannon {
 
   async start() {
     console.log("WEBHOOK CANNON STARTED");
-    await this.queue.consume(this.handleQueueMsg.bind(this));
+    await this.queue.consume("webhooks", this.handleQueueMsg.bind(this));
+    await this.queue.consume("events", this.handleEventsQueue.bind(this));
+  }
+
+  async handleEventsQueue (data: ConsumeMessage) {
+    let message;
+    try {
+      message = JSON.parse(data.content.toString());
+      console.log("events: got message", message);
+    } catch (err) {
+      console.log("events: error parsing message", err);
+      this.queue.ack(data);
+      return;
+    }
+    let event = message;
+
+    try {
+      const { data: webhooksList } = await this.db.webhook.listSubscribed(
+        event.userId,
+        event.event
+      );
+  
+      console.log("webhooks : ", webhooksList);
+      let stream = await this.db.stream.get(event.streamId);
+      if (!stream) {
+        // if stream isn't found. don't fire the webhook, log an error
+        return console.error(
+          `webhook Cannon: onTrigger: Stream Not found , streamId: ${event.streamId}`
+        );
+      }
+      // basic sanitization.
+      let sanitized = this.db.stream.addDefaultFields(
+        this.db.stream.removePrivateFields({ ...stream })
+      );
+      delete sanitized.streamKey;
+  
+      let user = await this.db.user.get(event.userId);
+      if (!user) {
+        // if user isn't found. don't fire the webhook, log an error
+        return console.error(
+          `webhook Cannon: onTrigger: User Not found , userId: ${event.userId}`
+        );
+      }
+
+      const responses = await Promise.all(
+        webhooksList.map(async (webhook, key) => {
+          try {
+            this.queue.publish("webhooks.triggers", {
+              id: uuid(),
+              event: event,
+              stream: sanitized,
+              user,
+              webhook
+            })
+          } catch (error) {
+            console.log("Error firing single url webhook trigger", error);
+          }
+        })
+      );
```

Comment:
I think we should avoid `ack`ing the event in case of an exception here. If we failed to send the event to rabbitmq, it most likely means that `rabbitmq` is going through some issues so we can just `nack` the message (after a `setTimeout` just in case?) to retry soon since none of the previous steps of fetching stream, user, etc had any issues (which would be the ones for us to drop the event on any error).

I think something like this would do it:
```ts
      try {
        const responses = await Promise.all(
          webhooksList.map((webhook) =>
              this.queue.publish("webhooks.triggers", {
                id: uuid(),
                event: event,
                stream: sanitized,
                user,
                webhook
              }))
        );
      } catch (err) {
         console.log("Error firing single webhook trigger events", err);
         setTimeout(10000, () => this.queue.nack(data));
         return
      }
```